src/pages/index.js

```diff
@@ -3,95 +3,157 @@ import Card from "/src/components/Card.js";
-    validator.enableValidation();
-  });
-};
-const createCard = (data) => {
-  const cardElement = new Card({
-    data: data,
-    cardSelector: cardTemplate,
-    handleCardClick: () => {
-      popupImage.open(data);
+Promise.all([api.getUserInformation(), api.getInitialCards()]).then(
+  ([userData, initialCards]) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserProfileImage(userData);
+    const apiCardSection = new Section(
+      {
+        items: initialCards,
+        renderer: addCard,
+      },
+      constants.cardGallery
+    );
+    apiCardSection.renderItems();
+  }
+);
+// Create Card + Add Card Functions
+const createCard = (cardData) => {
+  const cardElement = new Card(
+    cardData,
+    {
+      handleCardClick: () => popupImage.open(cardData),
+      handleDeleteClick: () => {
+        deleteModalNew.open();
+        deleteModalNew.submitForm(() => {
+          api
+            .deleteCard(cardData._id)
+            .then(() => {
+              deleteModalNew.close();
+              cardElement.removeCard();
+            })
+            .catch((err) => console.error(err));
+        });
+      },
+      handleLikeClick: (card) => {
+        // debugger;
+        if (!cardData.isLiked) {
+          api
+            .addLike(card)
+            .then((res) => {
+              cardElement.setLike(res);
+              console.log("add");
+              console.log(res.isLiked);
+            })
+            .catch((err) => console.error(err));
+        } else {
+          api
+            .deleteLike(card)
+            .then((res) => {
+              cardElement.setLike(res.isLiked);
+              console.error("delete");
+              console.log(res.isLiked);
+            })
+            .catch((err) => console.error(err));
+        }
+      },
     },
-  });
+    constants.cardTemplate
+  );
   return cardElement.getView();
 };
-const addCard = (data) => {
-  const card = createCard(data);
+const addCard = (cardData) => {
+  const card = createCard(cardData);
   cardSection.setItem(card);
 };
-
-const popupImage = new PopupWithImage(
+// Section instance to use methods
+const cardSection = new Section(
   {
-    popupImage: previewImage,
-    popupImageText: previewImageText,
+    items: [],
+    renderer: () => {},
   },
-  previewImageModal
+  constants.cardGallery
 );
-
-const userInfo = new UserInfo({
-  userName: profileName,
-  userJob: profileJob,
-});
-
-const cardSection = new Section(
+// Popup Instantiations
+const popupImage = new PopupWithImage(
   {
-    items: initialCards,
-    renderer: addCard,
+    popupImage: constants.previewImage,
+    popupImageText: constants.previewImageText,
   },
-  profileGallery
+  constants.previewImageModal
 );
-
 const editModalNew = new PopupWithForm(
   {
-    popup: editModal,
+    popup: constants.editModal,
     handleFormSubmit: (editModalData) => {
-      userInfo.setUserInfo(editModalData);
-      editModalNew.close();
+      api
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```